ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/UpdateNormalizer.scala

```diff
@@ -67,45 +69,17 @@ object RecordTimeNormalizer extends UpdateNormalizer {
   }
 }
 
-/** Rejection reason strings should always be ignored as they can change arbitrarily. */
-object RejectionReasonDescriptionNormalizer extends UpdateNormalizer {
-  override def normalize(update: Update): Update = update match {
-    case commandRejected @ Update.CommandRejected(_, _, reason) =>
-      val newReason = reason match {
-        case RejectionReasonV0.Disputed(_) =>
-          RejectionReasonV0.Disputed("")
-        case RejectionReasonV0.Inconsistent(_) =>
-          RejectionReasonV0.Inconsistent("")
-        case RejectionReasonV0.InvalidLedgerTime(_) =>
-          RejectionReasonV0.InvalidLedgerTime("")
-        case RejectionReasonV0.PartyNotKnownOnLedger(_) =>
-          RejectionReasonV0.PartyNotKnownOnLedger("")
-        case RejectionReasonV0.ResourcesExhausted(_) =>
-          RejectionReasonV0.ResourcesExhausted("")
-        case RejectionReasonV0.SubmitterCannotActViaParticipant(_) =>
-          RejectionReasonV0.SubmitterCannotActViaParticipant("")
-        // don't map non V0 errors
-        case x => x
-      }
-      commandRejected.copy(reason = newReason)
-    case _ => update
-  }
-}
-
-/** Rejection reasons `Disputed`, `InvalidLedgerTime` and `Inconsistent` are being conflated
-  *  as their usage has been re-designed in SDK 1.16.0-snapshot.20210727.7476.0.b5e9d861 [KVL-1015].
+/** The v1 to v2 migration makes it difficult to validate the rejection status
   */
-object CommandRejectionRedesignNormalizer extends UpdateNormalizer {
+object RejectionReasonNormalizer extends UpdateNormalizer {
   override def normalize(update: Update): Update = update match {
-    case commandRejected @ Update.CommandRejected(_, _, reason) =>
-      val newReason = reason match {
-        case RejectionReasonV0.Disputed(description) =>
-          RejectionReasonV0.Inconsistent(description)
-        case RejectionReasonV0.InvalidLedgerTime(description) =>
-          RejectionReasonV0.Inconsistent(description)
-        case _ => reason
-      }
-      commandRejected.copy(reason = newReason)
+    case commandRejected @ Update.CommandRejected(_, _, _) =>
+      val newReason = FinalReason(
```

Comment:
This is a radical normalization. Can we do something more refined here, please? I.e., are there cases that we could possibly recognize and map between before vs. after?

Reply:
I agree that this is basically a nuke. 
I'm not that knowledgeable as to how exactly does the integrity checker work (my assumption is that we have exports of the ledger and we replay those using the current code and compare. If that's the case then we could in theory just drop this normalizer as it should be compatible without it. Will test this before doing so.

Reply:
There really aren't many; it would be a huge endeavor to create a mapping.

If we are to do this, I suggest we do it later.

Reply:
Agreed regarding fixing this normalizer in a follow-up PR to limit scope.